fixtures/git_content_repos.py

```diff
@@ -1,10 +1,28 @@
 import pytest
+import json
+import urllib
+import urllib.error
+import urllib.parse
+import urllib.request
 
 
 @pytest.fixture
-def git_content_repos():
-    """Returns the list of all github content repos
+def git_content_repos(headers_data):
+
+    """Returns the list of all github content repos (starting with osbooks-)
     """
-    repo = ["osbooks-university-physics", "osbooks-college-algebra-bundle"]
+
+    repos_url = "https://api.github.com/orgs/openstax/repos?per_page=100&type=private"
```

Comment:
It looks like there are currently 77 private repos. The problem I see with using `per_page=100` versus using paging is if we go over 100 (and I believe 100 is the max one can request using `per_page`), the fixture could start missing repos without any obvious signal that this is happening. I'd suggest adding the paging support now so the fixture works with larger numbers of repos. However, if we leave the implementation this way, perhaps we should at least throw an exception if 100 results are returned as that will tell us the test code likely needs to be updated to support more results?

Reply:
yeah. I was thinking the same and wanted your opinion on this. I am also worried about paging. If I add page=4 (where the 2 osbooks- collections can be found) and in the near future more repos will be added, will not some of them fall to page=5? Or maybe at per_page=100 it won't happen that soon...?

Reply:
It's hard to predict when we might cross 100 since the repos are org wide. I know we'll be adding more for the content repos which will all be private, but I don't know for sure that would put us over 100. 

However, you shouldn't need to use page numbers directly. Something like the following could do the trick as far as iterating all pages (note I'm using `per_page=25` so there will be multiple pages for demonstration / testing):

```python
url = 'https://api.github.com/orgs/openstax/repos?per_page=25&type=private'
resp = requests.get(url, headers=headers)
print(f"Got {len(resp.json())} items")
while resp.links.get('next'):
    next_url = resp.links['next']['url']
    print(f"Getting next page at {next_url}")
    resp = requests.get(next_url, headers=headers)
    print(f"Got {len(resp.json())} items")
```

The next page link is provided in the response header when available, so you can just grab it and iterate that way.